src/pyclass.rs

```diff
@@ -165,13 +167,23 @@ where
     // buffer protocol
     type_object.tp_as_buffer = to_ptr(<T as class::buffer::PyBufferProtocolImpl>::tp_as_buffer());
 
+    let (new, call, mut methods, attrs) = py_class_method_defs::<T>();
+
     // normal methods
-    let (new, call, mut methods) = py_class_method_defs::<T>();
     if !methods.is_empty() {
         methods.push(ffi::PyMethodDef_INIT);
         type_object.tp_methods = Box::into_raw(methods.into_boxed_slice()) as *mut _;
     }
 
+    // class attributes
+    if !attrs.is_empty() {
```

Comment:
Considering a recursive case(e.g.,
```rust
#[pymethods]
impl MyClass {
    #[classattr]
    fn foo() -> MyClass { ... } 
}
```
), could you please move this code after `PyType_Ready`?
Then an incomplete type object is never used.

Reply:
That was precisely one of my use cases indeed. But I think it’s still possible to return `MyOtherClass` where `MyOtherClass` is has not yet been initialized?

Reply:
> But I think it’s still possible to return MyOtherClass where MyOtherClass has not yet been initialized?

Yes, but I'm not sure there's no corner case :thinking:

Reply:
Is it safe to modify `type_object` after `PyType_Ready` ?

Reply:
I realized that it cannot be a problem since `PyCell::new` doesn't use the type object at all.

Reply:
I’ll add a few « recursive » test cases to be sure that we don’t break that assumption in the future.

Reply:
> Is it safe to modify `type_object` after `PyType_Ready` ?

no, it's not safe, according to the C API docs